handler/src/main/java/io/netty/handler/pcap/PcapWriter.java

```diff
@@ -94,7 +109,9 @@ public void close() throws IOException {
         } else {
             isClosed = true;
             outputStream.flush();
-            outputStream.close();
+            if (!sharedOutputStream) {
```

Comment:
An argument could be made, since the OutputStream is passed in from the outside, it should never be closed from the writer in the first place? Users might do more things with it after the handler is done with its work.

Reply:
Certainly the current PcapWriteHandler code could be used on non-concurrent instances, writing to the same OutputStream, if it were not closed. Or, one could reopen the file for appending, and not write the Global Header, and use the OutputStream to save another channel. I guess the 5.0 codebase has been forked, so these changes would need to be applied in whichever form to 5.0 as well.